src/sfdxProject.ts

```diff
@@ -4,18 +4,57 @@
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
-import { defaults } from '@salesforce/kit';
+import { sep as pathSep } from 'path';
 
 import { ConfigAggregator } from './config/configAggregator';
 import { ConfigFile } from './config/configFile';
 import { ConfigContents } from './config/configStore';
 
+import { defaults, env } from '@salesforce/kit';
 import { JsonMap } from '@salesforce/ts-types';
-import { SfdxError } from './sfdxError';
+import { SchemaValidator } from './schema/validator';
 import { resolveProjectPath, SFDX_PROJECT_JSON } from './util/internal';
+
+// @ts-ignore
+import projectJsonSchema = require('@salesforce/schema/sfdx-project-schema.json');
+import { SfdxError } from './sfdxError';
 import { sfdc } from './util/sfdc';
 
+export interface PackageDirDependency {
```

Comment:
Hmm, so how did you generate these? Via a tool? It would be great to keep them up-to-date with the schemas. Maybe the schemes package should also ship types....

Reply:
Obviously outside the scope of this, but something that's would be useful I would think.

Reply:
I generated them and massaged them a bit.  Yeah it would be nice to have better TS support with the schema lib.